api/db-cdr/generate-cdr/make-bq-data.sh

```diff
@@ -209,33 +209,80 @@ case when count_value > 0 then round(count_value/$person_count, 2)
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.domain_info\` d
+set d.all_concept_count = c.all_concept_count, d.standard_concept_count = c.standard_concept_count from
+(select c.domain_id as domain_id, COUNT(DISTINCT c.concept_id) as all_concept_count,
+SUM(CASE WHEN c.standard_concept IN ('S', 'C') THEN 1 ELSE 0 END) as standard_concept_count from
+\`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.concept\` c
+join \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.domain_info\` d2
+on d2.domain_id = c.domain_id
+and (c.count_value > 0 or c.source_count_value > 0)
+group by c.domain_id) c
+where d.domain_id = c.domain_id
+"
+
+# Set participant counts for each domain
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.domain_info\` d
+set d.participant_count = r.count_value from
+\`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.achilles_results\` r
+where r.analysis_id = 3000 and r.stratum_1 = CAST(d.concept_id AS STRING)
+and r.stratum_3 = d.domain_id
+and r.stratum_2 is null
+"
+
+##########################################
+# survey count updates                   #
+##########################################
+
+# Set the survey participant count on the concept
 bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
 "update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.concept\` c1
 set c1.count_value=count_val from
 (select count(distinct ob.person_id) as count_val,cr.concept_id_2 as survey_concept_id from \`${BQ_PROJECT}.${BQ_DATASET}.observation\` ob
 join \`${BQ_PROJECT}.${BQ_DATASET}.concept_relationship\` cr
-on ob.observation_source_concept_id=cr.concept_id_1 join \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.db_domain\` dbd
-on cr.concept_id_2=dbd.concept_id
-where dbd.db_type='survey' and dbd.concept_id is not null
+on ob.observation_source_concept_id=cr.concept_id_1 join \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.survey_module\` sm
+on cr.concept_id_2=sm.concept_id
 group by cr.concept_id_2)
 where c1.concept_id=survey_concept_id"
 
+# Set the participant count on the survey_module row
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.survey_module\` sm
+set sm.participant_count=c.count_value from
+`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.concept\` c
+where c.concept_id=sm.concept_id"
+
+# Set the question count on the survey_module row
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.survey_module\` sm
+set sm.question_count=num_questions from
+(select count(distinct cr.concept_id_1) num_questions, cr.concept_id_2 as survey_concept_id from
+`${BQ_PROJECT}.${BQ_DATASET}.concept_relationship\` cr
+  join `${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.survey_module\` sm2
+    on cr.concept_id_2 = sm2.concept_id
+where cr.relationship_id = 'Has Module'
```

Comment:
I'm looking how we solved this before but the question counts you end up with are inflated due to data. We need to see what the final data  is but joining on question concept where count_value > 0 helps some. I think add that here and we figure out later on prod data what we need to fix 

select count(distinct cr.concept_id_1) num_questions, cr.concept_id_2 as survey_concept_id from
`all-of-us-workbench-test.cdr20180921.concept_relationship` cr
  join `all-of-us-workbench-test.cdr20180921.survey_module` sm2 
    on cr.concept_id_2 = sm2.concept_id 
  join `all-of-us-workbench-test.cdr20180921.concept` qc 
    on cr.concept_id_1 = qc.concept_id 
where cr.relationship_id = 'Has Module' and (qc.count_value > 0 ) 
  group by survey_concept_id

Reply:
Also, this query would need to move to after get the question counts below.

Reply:
Done.

Reply:
Cool. Yeah i think this is good. Our api code does some stuff i think when it getsSurveyQuestions ... We just show the array size as the qustion count. Ill see how to make this number accurate later.